src/build123d/topology.py

```diff
@@ -9248,6 +9068,242 @@ def unwrap_topods_compound(
     return compound
 
 
+def get_top_level_topods_shapes(topods_shape: TopoDS_Shape) -> ShapeList[TopoDS_Shape]:
+    """
+    Retrieve the first level of child shapes from the shape.
+
+    This method collects all the non-compound shapes directly contained in the
+    current shape. If the wrapped shape is a `TopoDS_Compound`, it traverses
+    its immediate children and collects all shapes that are not further nested
+    compounds. Nested compounds are traversed to gather their non-compound elements
+    without returning the nested compound itself.
+
+    Returns:
+        ShapeList[TopoDS_Shape]: A list of all first-level non-compound child shapes.
+
+    Example:
+        If the current shape is a compound containing both simple shapes
+        (e.g., edges, vertices) and other compounds, the method returns a list
+        of only the simple shapes directly contained at the top level.
+    """
+    if topods_shape is None:
+        return ShapeList()
+
+    first_level_shapes = []
+    stack = [topods_shape]
+
+    while stack:
+        current_shape = stack.pop()
+        if isinstance(current_shape, TopoDS_Compound):
+            iterator = TopoDS_Iterator()
+            iterator.Initialize(current_shape)
+            while iterator.More():
+                child_shape = downcast(iterator.Value())
+                if isinstance(child_shape, TopoDS_Compound):
+                    # Traverse further into the compound
+                    stack.append(child_shape)
+                else:
+                    # Add non-compound shape
+                    first_level_shapes.append(child_shape)
+                iterator.Next()
+        else:
+            first_level_shapes.append(current_shape)
+
+    return ShapeList(first_level_shapes)
```

Comment:
building the raw `list` only to convert to `ShapeList` looks unecessary. Either use a generator and do `return ShapeList(first_level_shapes())` or do `first_level_shapes = ShapeList[...]()` and just `return first_level_shapes`